lib/http2.c

```diff
@@ -2026,6 +2026,8 @@ static ssize_t http2_send(struct connectdata *conn, int sockindex,
       nva[i].namelen = strlen((char *)nva[i].name);
     }
     else {
+      /* Lower case the header name for HTTP/2 */
+      Curl_strntolower((char *)hdbuf, hdbuf, strlen(hdbuf));
```

Comment:
This should probably instead be done after the `namelen` assignment and use that length. Avoids strlen() in vain and uses a better length. I *think* this is why test 1700 triggered valgrind.
~~~c
nva[i].namelen = (size_t)(end - hdbuf);
Curl_strntolower((char *)hdbuf, hdbuf, nva[i].namelen);

Reply:
Fixed